### ISSUE TITLE
"Invalid character" when using interpolation within directive

### ISSUE THREAD

**Issue by nonmanager** [terraform-linters/tflint#1572]
## Introduction

When a simple directive is written, such as in [this](https://developer.hashicorp.com/terraform/language/expressions/strings#directives) example, interpolation cannot be used. Using the `$` character within the scope of the directive causes the linter to fail with an error.

Sample code (tested with Terraform 1.2.9):

```hcl
terraform {
  required_version = ">= 1.2.9"
}

variable "myvar" {
  type    = string
  default = "dear"
}

output "example" {
  value = "Hello, %{if var.myvar != "cruel"}${var.myvar}%{endif} world!"
}
```
Terraform happily accepts this:
```
Outputs:

example = "Hello, dear world!"
```

## Expected Behavior

Running `tflint` should succeed. It is what happens with `v0.41.0`.

## Actual behavior

Running `tflint` (`v0.42.0` and `v0.42.1`) fails with the following error:
```
Failed to check ruleset; Failed to check `terraform_deprecated_index` rule: main.tf:11,50-51: Invalid character; This character is not used within the language.
```

## Step to Reproduce

Run `tflint` on the code given above.

**Comment by bendrucker** [comment]
Definitely a bug. Duplicate of #1571.
